src/github/credentials.ts

```diff
@@ -25,26 +27,49 @@ export class CredentialStore {
 			return this._octokits[remote.url];
 		}
 
-		if (this._configuration.host === remote.host && this._configuration.accessToken) {
-			this._octokits[remote.url] = Octokit({});
-			this._octokits[remote.url].authenticate({
-				type: 'token',
-				token: this._configuration.accessToken
-			});
-			return this._octokits[remote.url];
+		const webflow = new WebFlow(new VSCodeAppConfiguration(), remote.host);
+		let creds = this._configuration as IHostConfiguration;
```

Comment:
It seems we only need `IHostConfiguration` in `CredentialStore`, we can unrestrict the type of `this._configuration` to be `IHostConfiguration`.

Reply:
Oh, yeah, good catch, this was me carefully walking around your existing Configuration implementation by avoiding changing signatures. All the configuration stuff needs cleaning up :P